src/Surfnet/StepupMiddleware/ApiBundle/Identity/Projector/RaCandidateProjector.php

```diff
@@ -266,30 +260,37 @@ private function updateInstitutionCandidatesFromCollection(Institution $institut
         foreach ($raInstitutions as $raInstitution) {
             // add new identities
             $raSecondFactors = $this->raSecondFactorRepository->findByInstitution($raInstitution->getInstitution());
-            foreach ($raSecondFactors as $raSecondFactor) {
-                $identity = $this->identityRepository->find($raSecondFactor->identityId);
-                $identityId = new IdentityId($identity->id);
 
-                // check if persistent in ra listing
-                if ($this->raListingRepository->findByIdentityIdAndRaInstitution($identityId, $institution)) {
+            foreach ($raSecondFactors as $raSecondFactor) {
+                if (!$raSecondFactor->status->equals(SecondFactorStatus::vetted())) {
                     continue;
                 }
 
-                // create candidate if not exists
-                $candidate = $this->raCandidateRepository->findByIdentityIdAndRaInstitution($identityId, $institution);
-                if (!$candidate) {
-                    $candidate = RaCandidate::nominate(
-                        $identityId,
-                        $identity->institution,
-                        $identity->nameId,
-                        $identity->commonName,
-                        $identity->email,
-                        $institution
-                    );
+                $identity = $this->identityRepository->find($raSecondFactor->identityId);
+                if ($identity) {
+                    $identityId = new IdentityId($identity->id);
+
+                    // check if persistent in ra listing
+                    if ($this->raListingRepository->findByIdentityIdAndRaInstitution($identityId, $institution)) {
+                        continue;
+                    }
+
+                    // create candidate if not exists
+                    $candidate = $this->raCandidateRepository->findByIdentityIdAndRaInstitution($identityId, $institution);
```

Comment:
Line above is always false. Because of the continue statement

Reply:
If it's on line 275 it uses another repository (raListing vs raCandidate)?
If you mean something else could you please collaborate what you mean?